src/include/mutex.h

```diff
@@ -83,7 +93,17 @@ struct WT_COMPILER_TYPE_ALIGN(WT_CACHE_LINE_ALIGNMENT) __wt_spinlock {
 struct WT_COMPILER_TYPE_ALIGN(WT_CACHE_LINE_ALIGNMENT) __wt_spinlock {
 	wt_mutex_t lock;
 
-	const char *name;		/* Statistics: mutex name */
+	const char *name;		/* Mutex name */
+
+	/*
+	 * We track acquisitions and time spent waiting for some locks. For
+	 * performance reasons and to make it possible to write generic code
+	 * that tracks statistics for different locks, we store the offset
+	 * of the statistics fields to be updated during lock acquisition.
+	 */
+	int16_t stat_count;		/* acquisitions */
```

Comment:
Since the stats content is duplicated across ``__wt_spinlock`` implementations anyway, would it maybe make sense to create a single:

```
struct __wt_spinlock_stat {
    WT_SPINLOCK l;
    /* Stat offset fields */
    int16_t stat_count;
    int16_t stat_app_usecs;
    int16_t stat_int_usecs;
};
```

Then __wt_spin_lock_track would be something like:

```
int
__wt_spin_lock_track(WT_SESSION *sessoin, WT_SPIN_LOCK *lock) {
    WT_SPIN_LOCK_STAT *lock_stat;

    lock_stat = (WT_SPIN_LOCK_STAT *)lock;
   ...
}
```

It's a little bit horrible - so I don't mind if you don't like it. The other advantage is that non-tracked spin locks don't have the additional fields in them either.